### PR TITLE
CloudWebDavProvider

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
Reimplement cloud webdav provider over new FileSystemBlobStore (vfs), which in turn is implemented more directly over the jclouds.Blobstore.  Reducing dependencies on S3FS and implement new caching scheme.

#### Related Pull Requests
- 

#### Changes
copy
rename file
disable rename collection (for now at least)
Path case sensitive/insensitive support

### CODE REVIEW

File: api/src/org/labkey/api/webdav/FileSystemResource.java [hunk 1]
```diff
@@ -259,14 +260,32 @@ public String getAbsolutePath(User user)
...
     @Override
     public long copyFrom(User user, FileStream is) throws IOException
     {
-        File file = getFile();
+        FileLike file = getFileLike();
         boolean created = false;
         if (!file.exists())
         {
-            FileUtil.mkdirs(file.getParentFile(), AppProps.getInstance().isInvalidFilenameUploadBlocked());
+            var parent = file.getParent();
```
Comment (Sigmonia) [inline api/src/org/labkey/api/webdav/FileSystemResource.java]: should this be inlined?

File: api/src/org/labkey/api/data/TempTableTracker.java [hunk 2]
```diff
@@ -163,11 +163,13 @@ private void untrack()
...
         synchronized(createdTableNames)
         {
-            createdTableNames.remove(qualifiedName);
+            var ttt = createdTableNames.remove(qualifiedName);
             appendToLog("-" + schemaName + "\t" + tableName + "\n");
 
             if (createdTableNames.isEmpty() || System.currentTimeMillis() > lastSync + CacheManager.DAY)
                 synchronizeLog(false);
+
+            ttt.clear();
```
Comment (Sigmonia) [inline api/src/org/labkey/api/data/TempTableTracker.java:172]: is this needed?

File: api/src/org/labkey/api/webdav/FileSystemResource.java [hunk 3]
```diff
@@ -277,18 +296,26 @@ public long copyFrom(User user, FileStream is) throws IOException
...
         }
 
         try
         {
-            is.transferTo(file);
-            if (is.getLastModified() != null)
+            File ioFile = getFile();
+            if (null != ioFile)
+            {
+                is.transferTo(getFile());
```
Comment (Sigmonia) [inline api/src/org/labkey/api/webdav/FileSystemResource.java]: should this be `ioFile`?

File: api/src/org/labkey/vfs/AbstractFileSystemLike.java [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
 import java.net.URI;
 import java.nio.file.Path;
 
 abstract public class AbstractFileSystemLike implements FileSystemLike
 {
     final URI uri;
     final String scheme;
-    final String strUri;    // no trailing '/'
+    final boolean caseSensitive;
+    final String strUri; // no trailing '/'
```
Comment (Sigmonia) [inline api/src/org/labkey/vfs/AbstractFileSystemLike.java]: Is the comment still accurate? re: line 23

File: api/src/org/labkey/vfs/FileSystemLike.java [hunk 5]
```diff
@@ -52,6 +52,13 @@
  */
 public interface FileSystemLike
 {
+    // NOTE: a full webdav path consist of case-sensitive and case-sensitive parts
```
Comment (Sigmonia) [inline api/src/org/labkey/vfs/FileSystemLike.java]: is one of those supposed to be `case-insensitive`?

**Review by Sigmonia — APPROVED** [review]
Again, mostly minor nits

File: api/src/org/labkey/api/data/TempTableTracker.java (hunk 2 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/data/TempTableTracker.java:172]: OK, this is unrelated.  If we don't clear the weak reference, we will get called later at GC time to delete a table we've already taken care of.

File: api/src/org/labkey/api/webdav/FileSystemResource.java (hunk 3 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/webdav/FileSystemResource.java]: yes

**Pull request merged**